### PR TITLE
SD-1746 doc fixes and doc additions

### PR DESCRIPTION
**Pull request by CaveMa**
### Support Docs Update Checklist

Note: when you open a pull request, your changes are published to https://staging.sparkpost.com/docs.

Please take a moment to check each of these steps for your pull request:

- [ ] Give your pull request a meaningful name.
- [ ] Include a description of your changes in the pull request.
- [ ] _SparkPost staff only_: Check that your article appears [here](https://staging.sparkpost.com/docs) and looks correct.
- [ ] Check the links in your article.
- [ ] Use lowercase filenames.

### CODE REVIEW

File: momentum/3/3-reference/3-reference-conf-ref-prefer_ipv6_mxs.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+<a name="conf.ref.prefer_ipv6_mxs"></a>
+## Name
+
+prefer_ipv6_mxs — prefer IPv6 addresses over IPv4 addresses when they have the same priority
+
+## Synopsis
+
+`prefer_ipv6_mxs = true`
+
+<a name="idp23262528"></a>
```
Comment (cjarnold) [inline momentum/3/3-reference/3-reference-conf-ref-prefer_ipv6_mxs.md:15]: What does this name represent?

File: momentum/3/3-reference/3-reference-conf-ref-prefer_ipv6_mxs.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+## Description
+
+Use the `prefer_ipv6_mxs` to have momentum prefer IPv6 addresses over IPv4 addresses when they have the same priority:
```
Comment (cjarnold) [inline momentum/3/3-reference/3-reference-conf-ref-prefer_ipv6_mxs.md:18]: I think we could take Rich's description from https://sparkpost.atlassian.net/wiki/spaces/ENG/pages/169151212/RND+Prefer+IPv6+MXs#id-[RND]PreferIPv6MXs-prefer_ipv6_mxsConfigurationOption verbatim because it clarifies using some examples which I found helpful.  

> The option will cause Momentum to prefer any IPv6 addresses at a given MX priority level. E.g.: if there is an IPv6 and an IPv4 address at MX priority 10, then it will prefer the IPv6 address.

Momentum keeps track of whether hosts are up/down at a per binding level. This option does not override whether Momentum thinks a host is up or down. So, in this example, if the IPv6 address is considered to be down, then the system will use the IPv4 address.

There may be multiple priority levels defined for an MX - e.g.: 10 and 20 (lower = higher priority, i.e.: 10 highest priority here). The option only prefers IPv6 within a given priority level. If no IPv6 hosts are considered up at level 10, the system would try to use an IPv4 host that is up at priority level 10, before it would move on to trying the hosts at priority level 20.

File: momentum/3/3-reference/3-reference-conf-ref-prefer_ipv6_mxs.md [hunk 3]
```diff
@@ -0,0 +1,29 @@
+---
+title: "prefer_ipv6_mxs"
+description: "prefer IPv6 over IPv4 when they have the same priority"
+---
+
+<a name="conf.ref.prefer_ipv6_mxs"></a>
+## Name
+
+prefer_ipv6_mxs — prefer IPv6 addresses over IPv4 addresses when they have the same priority
```
Comment (cjarnold) [inline momentum/3/3-reference/3-reference-conf-ref-prefer_ipv6_mxs.md:9]: I would reword "when they have the same priority" to "at a given MX priority level" here and everywhere else that phrase is used.  It clarifies this is MX priority.

File: momentum/3/3-reference/3-reference-conf-ref-prefer_ipv6_mxs.md (hunk 1 above)
Comment (CaveMa) [reply momentum/3/3-reference/3-reference-conf-ref-prefer_ipv6_mxs.md:15]: removed isn't needed since the other webpages aren't refering to this.

File: momentum/4/config/index.md [hunk 4]
```diff
@@ -227,6 +227,7 @@ description: "This chapter provides the definitions of the configuration options
 | [use_sendfile](/momentum/4/config/ref-use-sendfile) | use sendfile() when sending mail |
 | [user](/momentum/4/config/ref-user) | security: user identity to assume after startup |
 | [xclient](/momentum/4/config/ref-xclient) | use the xclient extension to SMTP for outbound mail |
+| [prefer_ipv6_mxs](/momentum/4/config/ref-prefer_ipv6_mxs) | prefer IPv6 addresses over ipv4 addresses when they have the same priority |
```
Comment (RSeabrease) [inline momentum/4/config/index.md:230]: "at a given mx priority level" needed here too?

File: momentum/4/modules/ipv-6-lookup.md [hunk 5]
```diff
@@ -3,17 +3,17 @@ title: "ipv6_lookup – Multi-address-family MX Records"
 description: "Configuration Change This feature is available in Momentum 4 2 and later The ipv 6 lookup module supports multi address family MX records enabling A record lookups for I Pv 6 addresses This module keeps track of whether the A record query succeeded and if it did when getting the..."
 ---
 
-<a name="idp21948288"></a> 
+<a name="idp21948288"></a>
 
-**Configuration Change. ** This feature is available in Momentum 4.2 and later.
+**Configuration Change. ** This feature only needs to be enabled on 4.2.x versions. 4.3.0 and higher already has it built in.
```
Comment (RSeabrease) [inline momentum/4/modules/ipv-6-lookup.md:8]: I think the wording here could use a little improvement. 
Does this mean that the module only needs to be enabled in the configuration on 4.2.x, and is automatically enabled in 4.3.0 and higher? Was the feature made available in 4.2, but 4.2.x needs the module to be loaded and 4.3.0 does not need to load the module? I think removing when the feature was made available could confuse some customers. Maybe leave in when the feature was made available, then clarify how to enable it.

File: momentum/4/config-options-summary.md [hunk 6]
```diff
@@ -11,11 +11,12 @@ If an option has a default value, it is shown in the `Default` column followed b
 
 The `Version` column indicated the version(s) of Momentum that support the option.
 
-<a name="all-options-table"></a> 
+<a name="all-options-table"></a>
 
 
 | Option/Description | Type | Default | Version | Scopes |
 | --- | --- | --- | --- | --- |
+| [prefer_ipv6_mxs](/momentum/4/config/ref-prefer_ipv6_mxs) - Prefer IPv6 addresses over IPv4 addresses at a given MX priority level | sending | false | 4.0 and later | global |
```
Comment (RSeabrease) [inline momentum/4/config-options-summary.md:19]: everything else in this list looks to be in alphabetic order. Can you move this down to follow that setup?

File: momentum/3/3-reference/options-summary.md [hunk 7]
```diff
@@ -6,11 +6,12 @@ description: "This section displays all configuration file options visible in th
...
 
-<a name="All-options-table"></a> 
+<a name="All-options-table"></a>
 
 
 | Option/Description | Type | Default | Version | Scopes |
 | --- | --- | --- | --- | --- |
+| [prefer_ipv6_mxs](/momentum/3/3-reference/3-reference-conf-prefer_ipv6mxs) - Prefer IPv6 addresses over IPv4 addresses at a given MX priority level | sending | false | 3.6.12 | global |
```
Comment (RSeabrease) [inline momentum/3/3-reference/options-summary.md:14]: Same thing here with the alphabetic order, can you move this down so it fits with the rest?

**Review by RSeabrease — CHANGES_REQUESTED** [review]

File: momentum/4/config/index.md (hunk 4 above)
Comment (CaveMa) [reply momentum/4/config/index.md:230]: good catch

File: momentum/4/modules/ipv-6-lookup.md (hunk 5 above)
Comment (CaveMa) [reply momentum/4/modules/ipv-6-lookup.md:8]: `**Configuration Change. ** This feature is available in Momentum 4.2 and later.
As of 4.3.0 users do not have to enable the feature only in 4.2`

Comment (RSeabrease) [reply momentum/4/modules/ipv-6-lookup.md:8]: `As of 4.3.0 users do not have to enable the module, but the module must be enabled in 4.2.x` sounds a little better.

File: momentum/3/3-reference/3-reference-conf-ref-prefer_ipv6_mxs.md [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+
+`prefer_ipv6_mxs = true`
+
+## Description
+
+The option will cause Momentum to prefer any IPv6 addresses at a given MX priority level. E.g.: if there is an IPv6 and an IPv4 address at MX priority 10, then it will prefer the IPv6 address.
+
+Momentum keeps track of whether hosts are up/down at a per binding level. This option does not override whether Momentum thinks a host is up or down. So, in this example, if the IPv6 address is considered to be down, then the system will use the IPv4 address.
+
+There may be multiple priority levels defined for an MX - e.g.: 10 and 20 (lower = higher priority, i.e.: 10 highest priority here). The option only prefers IPv6 within a given priority level. If no IPv6 hosts are considered up at level 10, the system would try to use an IPv4 host that is up at priority level 10, before it would move on to trying the hosts at priority level 20.
```
Comment (hlvass) [inline momentum/3/3-reference/3-reference-conf-ref-prefer_ipv6_mxs.md:21]: 10 and 20 (lower = higher priority, i.e.: 10 highest priority here) is a bit confusing

Comment (CaveMa) [reply momentum/3/3-reference/3-reference-conf-ref-prefer_ipv6_mxs.md:21]: you think we should delete it?

Comment (hlvass) [reply momentum/3/3-reference/3-reference-conf-ref-prefer_ipv6_mxs.md:21]: I think it's probably valuable to keep; it just seems like it's missing a couple words. Reading it a few times I get it and I think if you say "10 and 20 ( lower number = higher priority, i.e.: 10 being the higher priority here)" that would have made sense on the first read through.

Comment (CaveMa) [reply momentum/3/3-reference/3-reference-conf-ref-prefer_ipv6_mxs.md]: updated.

**Review by RSeabrease — APPROVED** [review]

**Review by cjarnold — APPROVED** [review]
